vector/src/main/java/im/vector/app/features/home/room/detail/search/SearchResultController.kt

```diff
@@ -148,6 +148,13 @@ class SearchResultController @Inject constructor(
      */
     private fun setHighLightedText(text: String, highlights: List<String>): Spannable? {
         val wordToSpan: Spannable = SpannableString(text)
+
+        // Some backends don't return highlights at all (this is the case of Synapse with the SQLite storage).
+        // In this case, don't attempt to filter based on the highlights, to avoid erasing all the results...
+        if (highlights.isEmpty()) {
+            return wordToSpan
+        }
```

Comment:
This will work, hence another possible fix would have been to just remove the `var found` below.
Better it's maybe clearer like this.